shell/platform/common/cpp/text_range.h

```diff
@@ -47,6 +47,23 @@ class TextRange {
   // Returns true if the range is of length 0.
   bool collapsed() const { return base_ == extent_; }
 
+  // Returns true if the base is greater than the extent.
+  bool reversed() const { return base_ > extent_; }
+
+  // Returns true if |position| is contained within the range.
+  bool Contains(size_t position) const {
+    return position >= start() && position < end();
```

Comment:
Why is this non-inclusive, but the `range` version below is inclusive? Doesn't this mean that `Contains(some_collapsed_range)` and `Contains(some_collapsed_range.position())` would return different results?

Reply:
Thanks for spotting this. I'd originally written them both as <, realised we need to cover cursor positions after the laste text position, then went back and fixed the range-based version but didn't fix the position version or the tests. Fixed now.